### PR TITLE
Provide an endpoint to get plugin presets

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/area core
/milestone 2.3.x

#### What this PR does / why we need it:

This PR provide a new endpoint to get plugins presets. Please see the result:

```bash
❯ curl -s -u admin:admin 'http://127.0.0.1:8090/apis/api.console.halo.run/v1alpha1/plugin-presets'
```
```json
[
  {
    "spec": {
      "displayName": "评论组件",
      "version": "1.2.0",
      "author": {
        "name": "Halo OSS Team",
        "website": "https://github.com/halo-dev"
      },
      "logo": "https://halo.run/logo",
      "pluginDependencies": {},
      "homepage": "https://github.com/halo-sigs/plugin-comment-widget",
      "description": "为用户前台提供完整的评论解决方案",
      "license": [
        {
          "name": "MIT"
        }
      ],
      "requires": ">=2.2.0",
      "enabled": true
    },
    "status": {
      "phase": "RESOLVED"
    },
    "apiVersion": "plugin.halo.run/v1alpha1",
    "kind": "Plugin",
    "metadata": {
      "name": "PluginCommentWidget"
    }
  },
  {
    "spec": {
      "displayName": "搜索组件",
      "version": "1.0.0",
      "author": {
        "name": "Halo OSS Team",
        "website": "https://github.com/halo-dev"
      },
      "logo": "https://halo.run/logo",
      "pluginDependencies": {},
      "homepage": "https://github.com/halo-sigs/plugin-search-widget",
      "description": "为 Halo 2.0 提供统一的搜索组件，方便主题端使用。",
      "license": [
        {
          "name": "MIT"
        }
      ],
      "requires": ">=2.0.0",
      "enabled": true
    },
    "status": {
      "phase": "RESOLVED"
    },
    "apiVersion": "plugin.halo.run/v1alpha1",
    "kind": "Plugin",
    "metadata": {
      "name": "PluginSearchWidget"
    }
  }
]
```

#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/3120

#### Special notes for your reviewer:

1. Try to download some plugins from <https://github.com/halo-sigs/awesome-halo>.
2. Put those jars into folder `src/main/resources/presets/plugins/`.
3. Get all plugins presets using the command below:
    
    ```bash
    # You might change your own username and password by yourself.
    curl -s -u admin:admin 'http://127.0.0.1:8090/apis/api.console.halo.run/v1alpha1/plugin-presets'
    ```

#### Does this PR introduce a user-facing change?

```release-note
提供预设插件功能
```

### CODE REVIEW

File: src/main/java/run/halo/app/core/extension/endpoint/PluginEndpoint.java [hunk 1]
```diff
@@ -169,9 +172,19 @@ public RouterFunction<ServerResponse> endpoint() {
                     .response(responseBuilder()
                         .implementation(ConfigMap.class))
             )
+            .GET("plugin-presets", this::listPresets, builder -> {
```
Comment (guqing) [inline src/main/java/run/halo/app/core/extension/endpoint/PluginEndpoint.java:175]: 这个 API 是否需要添加权限到角色模板，不添加的话只有超级管理员可以访问到

Comment (JohnNiang) [reply src/main/java/run/halo/app/core/extension/endpoint/PluginEndpoint.java:175]: 这里可能需要加到“插件管理”角色模板中。

**Comment by JohnNiang** [comment]
/hold

当前仅仅提供了预设插件列表接口，但并未提供安装预设插件的功能。

**Comment by ruibaby** [comment]
推荐预设的插件：

- https://github.com/halo-sigs/plugin-comment-widget
- https://github.com/halo-sigs/plugin-search-widget
- https://github.com/halo-sigs/plugin-sitemap
- https://github.com/halo-sigs/plugin-feed

**Comment by JohnNiang** [comment]
> 推荐预设的插件：
> 
> * https://github.com/halo-sigs/plugin-comment-widget
> * https://github.com/halo-sigs/plugin-search-widget
> * https://github.com/halo-sigs/plugin-sitemap
> * https://github.com/halo-sigs/plugin-feed

推荐在 GitHub Workflow 中下载固定版本的插件到对应的目录。

**Review by ruibaby — APPROVED** [review]
/approve

可以配合 https://github.com/halo-dev/console/pull/894 测试。

**Review by LIlGG — APPROVED** [review]
/lgtm

**Pull request merged**